cytotable/convert.py

```diff
@@ -69,8 +71,39 @@ def _get_table_columns_and_types(source: Dict[str, Any]) -> List[Dict[str, str]]
         WHERE segment_type != 'VALIDITY';
         """
 
-    # perform the query and create a list of dictionaries with the column data for table
-    return _duckdb_reader().execute(select_query).arrow().to_pylist()
+    # attempt to read the data to parquet from duckdb
+    # with exception handling to read mixed-type data
+    # using sqlite3 and special utility function
+    try:
+        # isolate using new connection to read data with chunk size + offset
+        # and export directly to parquet via duckdb (avoiding need to return data to python)
+        # perform the query and create a list of dictionaries with the column data for table
+        return (
+            _duckdb_reader()
+            .execute(select_query.replace("&select_source", select_source))
+            .arrow()
+            .to_pylist()
+        )
+
+    except duckdb.Error as e:
+        # if we see a mismatched type error
+        # run a more nuanced query through sqlite
+        # to handle the mixed types
+        if "Mismatch Type Error" in str(e) and source_type == ".sqlite":
+            arrow_data_tbl = _sqlite_mixed_type_query_to_parquet(
+                source_path=str(source["source_path"]),
+                table_name=str(source["table_name"]),
+                chunk_size=5,
```

Comment:
This chunk size seems small, also, it is hardcoded. Just checking that this is ok.

Reply:
Thanks for the check on this - `chunk_size` is set to 5 here so as to use a similar SQL `LIMIT` on row results from the table for data type inference. The nomenclature of the parameter stems from how the function is used when extracting "chunks" of data later on. We use this constraint on the dataset in this case so as to enable "good enough" data type inferencing while balancing performance (not reading the full table will increase the speed by which the task completes). I added some code comments to increase the description of what's happening in these areas.